addons/im_livechat/tests/test_get_operator.py

```diff
@@ -0,0 +1,231 @@
+            ),
+        )
+
+    def test_operator_in_call_no_more_than_two_chats(self):
+        fist_operator = self._create_operator("fr_FR", "FR")
+        second_operator = self._create_operator("fr_FR", "FR")
+        livechat_channel = self.env["im_livechat.channel"].create(
+            {
+                "name": "Livechat Channel",
+                "user_ids": [fist_operator.id, second_operator.id],
+            }
+        )
+        channel_1_info = self.make_jsonrpc_request(
+            "/im_livechat/get_session",
+            {
+                "anonymous_name": "Visitor 1",
+                "channel_id": livechat_channel.id,
+                "previous_operator_id": fist_operator.partner_id.id,
+                "persisted": True,
+            },
+        )
+        channel_2_info = self.make_jsonrpc_request(
+            "/im_livechat/get_session",
+            {
+                "anonymous_name": "Visitor 2",
+                "channel_id": livechat_channel.id,
+                "previous_operator_id": fist_operator.partner_id.id,
+                "persisted": True,
+            },
+        )
+        # Only channels with messages within the last 30 minutes are considered
+        # as active, mark them as active now.
+        self.env["mail.message"].create(
+            [
+                {
+                    "model": "discuss.channel",
+                    "res_id": channel_1_info["id"],
+                    "body": "Hello, I'm visitor 1",
+                },
+                {
+                    "model": "discuss.channel",
+                    "res_id": channel_2_info["id"],
+                    "body": "Hello, I'm visitor 2",
+                },
+            ]
+        )
+        # Previous operator is not in a call so it should be available, even if
+        # he already has two ongoing chats.
+        self.assertEqual(
+            fist_operator, livechat_channel._get_operator(previous_operator_id=fist_operator.partner_id.id)
+        )
+        self.env["discuss.channel.rtc.session"].create(
+            {
+                "channel_id": channel_1_info["id"],
+                "channel_member_id": self.env["discuss.channel.member"]
+                .search([["partner_id", "=", fist_operator.partner_id.id], ["channel_id", "=", channel_1_info["id"]]])
+                .id,
+            }
+        )
+        # Previous operator is in a call so it should not be available anymore.
+        self.assertEqual(
+            second_operator, livechat_channel._get_operator(previous_operator_id=fist_operator.partner_id.id)
+        )
+
+    def test_find_less_active_operator_ids(self):
+        # The one with less chats should be returned
+        self.assertEqual(
+            [1],
+            self.env["im_livechat.channel"]._find_less_active_operator_ids(
+                {
+                    1: {"count": 1, "in_call": False},
+                    2: {"count": 2, "in_call": False},
+                }
+            ),
+        )
```

Comment:
I think unit tests on `_get_operator` have some value, the ones on `_find_less_active_operator_ids` don't.
We should avoid making tests that just tests functions, because they don't necessarily translate to good working of the application (what if the function is unused, or used incorrectly, etc.), and can even prevent us to make the necessary code refactoring due to reliance on implementation details.

- `_get_operator` is pretty close to functional need, so it's good
- `_find_less_active_operator_ids ` goes to much into implementation details.

Could you make a similar coverage while only using `_get_operator`?

Reply:
Unfortunately I can't: `_get_operator` will return *one* of the available operators so its pure randomness. `_find_less_active_operator_ids` has a pretty complicated logic and I though testing it would give us more confidence in the future.

Reply:
Also writing the `_find_less_active_operator_ids` test highlighted flaws in the function that were fixed thanks of it. I think it adds some value.